tests/components/samsungtv/test_media_player.py

```diff
@@ -169,7 +174,7 @@ def mock_now():
 
 async def setup_samsungtv(hass, config):
     """Set up mock Samsung TV."""
-    await async_setup_component(hass, "media_player", config)
+    await samsungtv.async_setup(hass, config)
```

Comment:
Don't call any integration functions to set up the tests. Please still use `async_setup_component` but change domain and config domain to `"samsungtv`.

We may also need to add a mock entry.

Reply:
@MartinHjelmare When I do this I get a stack trace and I don't understand why:

```
ERROR    homeassistant.setup:setup.py:182 Error during setup of component samsungtv
Traceback (most recent call last):
  File "./home-assistant/homeassistant/setup.py", line 172, in _async_setup_component
    hass, processed_config
  File "./home-assistant/homeassistant/components/samsungtv/__init__.py", line 48, in async_setup
    DOMAIN, context={"source": "user"}, data=entry_config
  File "./home-assistant/homeassistant/data_entry_flow.py", line 73, in async_init
    return await self._async_handle_step(flow, flow.init_step, data)
  File "./home-assistant/homeassistant/data_entry_flow.py", line 154, in _async_handle_step
    result = await self._async_finish_flow(flow, dict(result))
  File "./home-assistant/homeassistant/config_entries.py", line 619, in _async_finish_flow
    await self.async_setup(entry.entry_id)
  File "./home-assistant/homeassistant/config_entries.py", line 500, in async_setup
    self.hass, entry.domain, self._hass_config
  File "./home-assistant/homeassistant/setup.py", line 45, in async_setup_component
    return await setup_tasks[domain]  # type: ignore
RuntimeError: Task cannot await on itself: <Task pending coro=<_async_setup_component() running at ./home-assistant/homeassistant/setup.py:172> cb=[<TaskWakeupMethWrapper object at 0x7ff8e4d00a10>()]>

```

Reply:
Well in your stack trace `config_entries` is calling `async_setup_component`, which is wrong.

Reply:
I'm a bit confused. Isn't this what @MartinHjelmare asked me to do?

Reply:
Please push what you have changed so we can look at that. It's a waste of time to speculate about this stack trace, when we can look at the source too.

Reply:
@MartinHjelmare It's exactly like in your marked rows. With `samsungtv.async_setup` everything is working fine, with `async_setup_component` I get this stack trace above. Well, you said to use the latter but @balloob states it's wrong.

Reply:
Here's the bug:
https://github.com/home-assistant/home-assistant/pull/28306/files#discussion_r362639162

Reply:
The solution was simple but the problem not obvious, at least for me...